__tests__/tests/costumes.test.js

```diff
@@ -0,0 +1,325 @@
+}
+
+beforeEach( async () => {
+    await createTables(pool);
+
+    await createCostume(pool, getBallroomGown());
+    await createCostume(pool, getButtlessChaps());
+    await createCostume(pool, getBonnet());
+})
+
+// Disconnect from postgres database after all tests done
+afterAll(async () => {
+    await pool.end()
+})
+
+describe("createTables adapter", () => {
+    it("should create a table", async () => {
+        const costumes = await pool.query(`
+            SELECT * FROM costumes;
+        `)
+
+        expect(costumes).toBeTruthy();
+    })
+})
+
+describe("createCostume adapter", () => {
+    it("should create a new entry with correct values", async () => {
+        const {rows: [chapsFromDatabase]} = await pool.query(`
+            SELECT * FROM costumes WHERE name='buttless chaps';
+        `);
+        
+        expect(matchesDatabase(getButtlessChaps(), chapsFromDatabase)).toBe(true);
+    })
+
+    it("should create multiple entries when called multiple times", async () => {
+        const {rows: [bonnetFromDatabase]} = await pool.query(`
+            SELECT * FROM costumes WHERE name='bonnet';
+        `);
+
+        const {rows: [chapsFromDatabase]} = await pool.query(`
+            SELECT * FROM costumes WHERE name='buttless chaps';
+        `);
+
+        expect(matchesDatabase(getBonnet(), bonnetFromDatabase)).toBe(true);
+        expect(matchesDatabase(getButtlessChaps(), bonnetFromDatabase)).toBe(false);
+        expect(matchesDatabase(getButtlessChaps(), chapsFromDatabase)).toBe(true);
+        expect(matchesDatabase(getBonnet(), chapsFromDatabase)).toBe(false);
+
+    })
+
+    it("should throw an error if not given enough arguments", async () => {
+        expect.hasAssertions();
+        try {
+            await createCostume(pool, getBonnetMissingArg())
+        } catch (e) {
+            expect(e.name).toMatch('error');
+            expect(e.code).toMatch('23502');
+        }
+    })
+
+    it("should throw an error if given the wrong argument type", async () => {
+        expect.hasAssertions();
+        try {
+            await createCostume(pool, getGownWithWrongType())
+        } catch (e) {
+            console.log(e);
+            expect(e.name).toMatch('error');
+        }
+    })
+
+    it("should throw an error if argument does not follow CHECK constraint", async () => {
+        expect.hasAssertions();
+        try {
+            await createCostume(pool, getGownWithWrongCategory())
+        } catch (e) {
+            expect(e.name).toMatch('error');
+        }
+    })
+
+    it("should throw an error if argument does not follow VARCHAR length constraint", async () => {
+        expect.hasAssertions();
+        try {
+            await createCostume(pool, getGownWithLongSize())
+        } catch (e) {
+            expect(e.name).toMatch('error');
+        }
+    })
+})
```

Comment:
I would actually argue against this category of tests

If you consider what it is that is being tested here, it boils down to whether or not you set the correct constraints when you created the tables (because the only other thing being tested is whether postgres is working as expected)

It's a bit on the line, but in most cases, I would say don't write a test that just check a config, and in this case, how you create the tables feels more like a config than like business logic. In other-words test code, not which options you selected.

Reply:
Makes sense. How does one test if the configurations are working? Or do you?

Reply:
Honestly, often you wouldn't write tests specifically for that. Typically you can think of the configs almost like implementation details that should be caught by the combination of testing the features that actually use them - and end-to-end smoke-tests that confirm that nothing major is busted.